pkg/kubectl/kubectl.go

```diff
@@ -77,8 +78,21 @@ type ShortcutExpander struct {
 // VersionAndKindForResource implements meta.RESTMapper. It expands the resource first, then invokes the wrapped
 // mapper.
 func (e ShortcutExpander) VersionAndKindForResource(resource string) (defaultVersion, kind string, err error) {
-	resource = expandResourceShortcut(resource)
+	exp := "experimental"
+	parts := strings.Split(resource, "/")
+	// group/kind or group/kind-shortcut
+	if len(parts) > 1 {
+		resource = fmt.Sprintf("%s/%s", parts[0], expandResourceShortcut(parts[1]))
+	}
+	// kind or kind-shortcut
+	if len(parts) == 1 {
+		resource = expandResourceShortcut(resource)
+	}
 	defaultVersion, kind, err = e.RESTMapper.VersionAndKindForResource(resource)
+	// kind that should be experimental/kind
+	if group, err := e.GroupForResource(resource); len(parts) == 1 && err == nil && group == exp {
```

Comment:
How about checking len(parts)==1 before calling GroupForResource to save some function calls?